spec/controllers/posts_controller_spec.rb

```diff
@@ -317,14 +341,15 @@
         expect(assigns(:written).user).to eq(user)
         expect(assigns(:post)).to eq(assigns(:written))
         expect(assigns(:page_title)).to eq('Previewing: test')
+        expect(assigns(:author_ids)).to match_array([user.id, coauthor.id])
 
         # tags
-        expect(assigns(:post).settings.size).to eq(3)
-        expect(assigns(:post).content_warnings.size).to eq(3)
-        expect(assigns(:post).labels.size).to eq(3)
-        expect(assigns(:post).settings.map(&:id_for_select)).to match_array([setting1.id, setting2.id, '_other'])
-        expect(assigns(:post).content_warnings.map(&:id_for_select)).to match_array([warning1.id, warning2.id, '_other'])
-        expect(assigns(:post).labels.map(&:id_for_select)).to match_array([label1.id, label2.id, '_other'])
+        expect(assigns(:post).settings.size).to eq(0)
+        expect(assigns(:post).content_warnings.size).to eq(0)
+        expect(assigns(:post).labels.size).to eq(0)
```

Comment:
Since we're no longer storing these on the post until we actually save, we don't want to amend the relations on the post, just the local variables (just below this).